src/Controls/tests/Xaml.UnitTests/Issues/Maui20616.xaml.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using Microsoft.Maui.ApplicationModel;
+using Microsoft.Maui.Controls.Core.UnitTests;
+using Microsoft.Maui.Controls.Shapes;
+using Microsoft.Maui.Devices;
+using Microsoft.Maui.Dispatching;
+
+using Microsoft.Maui.Graphics;
+using Microsoft.Maui.UnitTests;
+using NUnit.Framework;
+
+namespace Microsoft.Maui.Controls.Xaml.UnitTests;
+
+public partial class Maui20616
+{
+    public Maui20616()
+    {
+        InitializeComponent();
+        BindingContext = new ViewModel20616<string> { Value = "Foo" };
+    }
+
+    public Maui20616(bool useCompiledXaml)
+    {
+        //this stub will be replaced at compile time
+    }
+
+    [TestFixture]
+    class Test
+    {
+        [SetUp]
+        public void Setup()
+        {
+            Application.SetCurrentApplication(new MockApplication());
+            DispatcherProvider.SetCurrent(new DispatcherProviderStub());
+        }
+
+        [TearDown] public void TearDown() => AppInfo.SetCurrent(null);
+
+        [Test]
+        public void XDataTypeCanBeGeneric([Values(false, true)] bool useCompiledXaml)
+        {
+            if (useCompiledXaml)
+            {
+                MockCompiler.Compile(typeof(Maui20616));
+            }
+            else
+            {
+                new Maui20616(useCompiledXaml: false);
+            }
```

Comment:
the test should be 
```csharp
var page = new Maui20616(useCompiledXaml);
...```

and somehow assert that the TypedBinding has the correct datatype, or at least that it worked

the MockCompiler.Compile isn't necessary, unless you need to detect that it fails, or report warnings